frontends/ios/BitBoxApp/BitBoxApp/Bluetooth.swift

```diff
@@ -448,10 +461,63 @@ class BluetoothDeviceInfo: NSObject, MobileserverGoDeviceInfoInterfaceProtocol {
 
 class BluetoothReadWriteCloser: NSObject, MobileserverGoReadWriteCloserInterfaceProtocol {
     private let bluetoothManager: BluetoothManager
+    private var write_buffer = Data()
+    private let write_buffer_lock = NSLock()
+    private var writer_thread: Thread?
+    private let semaphore = DispatchSemaphore(value: 0)
 
     init(bluetoothManager: BluetoothManager) {
         self.bluetoothManager = bluetoothManager
     }
+    func start() {
+        writer_thread = Thread.init(target: self, selector: #selector(writer), object: nil)
+        writer_thread!.start()
+
+    }
+
+    @objc func writer(_ _: Any) {
```

Comment:
why a thread instead of writing it sequentially in `func write()`? isn't the write function already called from some goroutine, so it shouldn't be blocking the UI?